### PR TITLE
Add a new "Locale" extension to access the \Locale PHP class

### PR DESCRIPTION
**Pull request by mpdude**
Currently this only features a new `locale_primary_language` function to get the language for the current or a given locale.

### CODE REVIEW

**Comment by mpdude** [comment]
`ArrayTest` fails on PHP7?

File: lib/Twig/Extensions/Extension/Locale.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        return array(
+            new Twig_SimpleFunction('locale_primary_language', 'twig_locale_primary_language'),
+        );
+    }
+}
+
+function twig_locale_primary_language($locale = null)
+{
+    $locale = $locale !== null ? $locale : Locale::getDefault();
```
Comment (SpacePossum) [inline lib/Twig/Extensions/Extension/Locale.php]: negative check can be swapped
`Locale` is in the `intl` package which is not required by this repo, nor by Twig itself, so it should be made a required ext. or be polyfilled

Comment (stof) [inline lib/Twig/Extensions/Extension/Locale.php]: this is not a required dependency of the whole package, as it contains several extension. But ext-intl (or the Symfony partial polyfill for the en locale) is indeed a dependency when using this extension, as it is precisely about integrating intl.

Comment (mpdude) [inline lib/Twig/Extensions/Extension/Locale.php]: Would a check in the constructor suffice?

Comment (mpdude) [inline lib/Twig/Extensions/Extension/Locale.php]: @stof / @SpacePossum is it better that way?

**Comment by mpdude** [comment]
The test failure is on PHP7 and I don't see how this is related.

**Comment by fabpot** [comment]
Closing this PR as this repository is going to be deprecated soon.

**Pull request closed without merging**